src/search_algo.hpp

```diff
@@ -430,7 +458,11 @@ seedLooksPromising(LocalDataHolder<TGlobalHolder> const & lH,
     // score the diagonal
     for (uint64_t i = 0; i < effectiveLength; ++i)
     {
-        s += lH.gH.scoringSchemePreScoring.score(qSeq[i], sSeq[i]);
+        if constexpr (fwd)
```

Comment:
Can't you do the same thing here as in the alignment and get rid of the extra template parameter? `auto & scoringScheme = fwd ? ...` before the loop?